### ISSUE TITLE
Problem with 1-loop matching onto Cegamma

### ISSUE THREAD

**Issue by MJKirk** [wilson-eft/wilson#95]
I believe there is a bug in the 1-loop matching onto the LEFT coefficient egamma.
A complete example is 
```python
from wilson import Wilson

w = Wilson({"phiq1_33": 1e-8}, scale=1000, eft="SMEFT", basis="Warsaw")
w.set_option("smeft_matching_order", 1)
res = w.match_run(scale=91.1876, eft="WET", basis="JMS")
print(res["egamma_12"])
```
which gives a value for `egamma_12` of about -4.5e-13. This value is sufficiently large to almost exceed the experimental bound on BR(mu->egamma) (a bound from LFV observables in smelli is how I picked this up).
Since I'm pretty confidant there is no way at purely quark coefficient (and a diagonal one at that!) can induce some LFV beyond that of the SM, this must be a bug.

I have (partially) narrowed down the problem.
```python 
wlow = Wilson({"phiD": -2.6e-9}, scale=91.1876, eft="SMEFT", basis="Warsaw")
wlow.set_option("smeft_matching_order", 1)
print(wlow.match_run(scale=91.1876, eft="WET", basis="JMS")["egamma_12"])
```
gives a value of `egamma_12` of about -4.6e-13, so roughly the same as the full result.
Looking at the 1-loop matching code in https://github.com/wilson-eft/wilson/blob/8931ea52cd8cedab3712a207a9bc02cdaa4e6151/wilson/match/smeft_loop.py, there are several instances of `C["phiD"]*einsum("p,r",one,me)`.
If I were a betting man, I'd say this is likely the same problem as #50.

In fact, if I apply the replacement 
`sed -i -e 's/einsum("p,r",one,me)/np.diag(me)/g'  wilson/match/smeft_loop.py` (as stolen from the patch that fixed that issue), then egamma_12 is no longer generated.

**Comment by MJKirk** [comment]
Oh, I forgot to mention. I did check this matching with the original notebooks from the  Dekens & Stoffer paper, and indeed `egamma` has some matching terms proportional to `phiD` and they go as delta_ij me_i, so the "corrected" code make sense.

**Comment by peterstangl** [comment]
@MJKirk thank you for reporting this! It indeed looks like a bug. Now that issue #55 is fixed, @jasonaebischerGIT and me want to cross check the full one-loop matching. In the meantime, would you like to provide your solution to fix this bug as a PR?
Btw., I noticed that in addition to `einsum("p,r",one,me)` in `egamma`, also `einsum("p,r",one,mu)` appears in `uG` and `ugamma` and comparing with the Dekens & Stoffer publication reveals that this should also be delta_ij mu_i.

**Comment by peterstangl** [comment]
This issue has been fixed by @MJKirk in PR #100. The bug fix is included in wilson v2.3.1.
